File: lab08/Sunrise/app/src/main/java/com/example/sunrise/MainActivity.java
package com.example.sunrise;

import android.os.Bundle;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.widget.ImageView;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ImageView sun = findViewById(R.id.sun);

        Animation animation = AnimationUtils.loadAnimation(this, R.anim.sun_rise);
        sun.startAnimation(animation);

        ImageView hourImageView = findViewById(R.id.hour_hand);

        Animation hourAnimation =
                AnimationUtils.loadAnimation(this, R.anim.hour_turn);

        hourImageView.startAnimation(hourAnimation);
    }
}

File: lab05/kotlin/AndroidLabs/app/src/main/java/com/example/toast_kotlin/MainActivity.kt
package com.example.toast_kotlin

import android.os.Bundle
import android.view.Gravity
import android.view.LayoutInflater
import android.view.View
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
    }

    fun showToast(view: View) {
        val inflater = LayoutInflater.from(this)
        val layout = inflater.inflate(R.layout.custom_layout,
            findViewById(R.id.toast_layout))

        val toast = Toast(applicationContext)
        toast.setGravity(Gravity.CENTER, 0, 0)
        toast.duration = Toast.LENGTH_LONG
        toast.view = layout
        toast.show()
    }
}